test/unit/puppetlabs/services/jruby/jruby_puppet_agents_test.clj

```diff
@@ -1,134 +0,0 @@
-                                    (deliver pool-state-swapped true)))]
-        ; borrow an instance so we know that the pool is ready
-        (jruby/with-jruby-puppet jruby-puppet jruby-service :retry-poison-pill-test)
-        (add-watch (:pool-state pool-context) :pool-state-watch pool-state-watch-fn)
-        (jruby-protocol/flush-jruby-pool! jruby-service)
-        ; wait until we know the new pool has been swapped in
-        @pool-state-swapped
-        ; wait until the flush is complete
-        (await (:pool-agent pool-context))
-        (let [old-pool-instance (jruby-internal/borrow-from-pool!*
-                                  jruby-internal/borrow-without-timeout-fn
-                                  old-pool)]
-          (is (jruby-schemas/retry-poison-pill? old-pool-instance)))))))
-
-(deftest with-jruby-retry-test-via-mock-get-pool
-  (testing "with-jruby-puppet retries if it encounters a RetryPoisonPill"
-    (tk-testutils/with-app-with-config
-      app
-      default-services
-      (-> (jruby-testutils/jruby-puppet-tk-config
-            (jruby-testutils/jruby-puppet-config {:max-active-instances 1})))
-      (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
-            real-pool     (-> (tk-services/service-context jruby-service)
-                              :pool-context
-                              (jruby-core/get-pool))
-            retry-pool    (JRubyPool. 1)
-            _             (->> retry-pool
-                              (RetryPoisonPill.)
-                              (.insertPill retry-pool))
-            mock-pools    [retry-pool retry-pool retry-pool real-pool]
-            num-borrows   (atom 0)
-            get-mock-pool (fn [_] (let [result (nth mock-pools @num-borrows)]
-                                    (swap! num-borrows inc)
-                                    result))]
-        (with-redefs [jruby-internal/get-pool get-mock-pool]
-          (jruby/with-jruby-puppet
-            jruby-puppet
-            jruby-service
-            :with-jruby-retry-test
-            (is (instance? JRubyPuppet jruby-puppet))))
-        (is (= 4 @num-borrows))))))
-
-(deftest next-instance-id-test
-  (let [pool-context (jruby-core/create-pool-context
-                       (jruby-testutils/jruby-puppet-config {:max-active-instances 8})
-                       jruby-testutils/default-profiler
-                       jruby-testutils/default-shutdown-fn)]
-    (testing "next instance id should be based on the pool size"
-      (is (= 10 (jruby-agents/next-instance-id 2 pool-context)))
-      (is (= 100 (jruby-agents/next-instance-id 92 pool-context))))
-    (testing "next instance id should wrap after max int"
-      (let [id (- Integer/MAX_VALUE 1)]
-        (is (= (mod id 8) (jruby-agents/next-instance-id id pool-context)))))))
-
-(deftest master-termination-test
-  (testing "Flushing the pool causes masters to be terminated"
```

Comment:
I wonder if it might still be worthwhile to have a test like this, that validates that the cleanup function we've provided actually works?